client/src/secret/mod.rs

```diff
@@ -269,16 +271,25 @@ impl SecretManageExt for SecretManager {
     async fn sign_transaction_essence(
         &self,
         prepared_transaction_data: &PreparedTransactionData,
+        time: Option<u32>,
     ) -> crate::Result<Unlocks> {
         match self {
```

Comment:
Maybe its better to do this here
``` 
let time = time.unwrap_or_else(|| {
        instant::SystemTime::now()
            .duration_since(instant::SystemTime::UNIX_EPOCH)
            .expect("time went backwards")
            .as_secs() as u32
    });
```
Instead of on the 3 separate spots inside `sign_transaction_essence``s

Reply:
That's not possible, because the function signature would then be different (if we change the other methods to only accept u32 then)?

Reply:
yea i was thinking 2 methods one with option and one normal, option has an impl and jsut calls normal. but maybe thats a bit overkill... i jsut cant stand duplicate code :D

Reply:
2 methods instead of one just to prevent this time not being duplicated seems like duplicating even more :sweat_smile: 
But I added unix_timestamp_now() so it's less duplicated, are you fine with that?